### PR TITLE
Fix/homepage scoring

### PR DESCRIPTION
**Pull request by WRadoslaw**
#245

### CODE REVIEW

File: src/utils/VideoRelevanceManager.ts [hunk 1]
```diff
@@ -19,66 +17,56 @@ export class VideoRelevanceManager {
...
+        
+        UPDATE video
+        SET video_relevance = COALESCE(topChannelVideo.maxScore, 1) 
+        FROM videos_with_weight as videoCte
+        LEFT JOIN top_channel_score as topChannelVideo on topChannelVideo.channelId = videoCte.cId and topChannelVideo.maxScore = videoCte.videoRelevance
+        WHERE video.id = videoCte.vId;
         `)
-      this.videosToUpdate.clear()
-    }
   }
```
Comment (ignazio-bovo) [inline src/utils/VideoRelevanceManager.ts:64]: Can you maybe break down this complex query into smaller pieces for better readability? Even with Copilot is a bit complex to decipher

**Review by ignazio-bovo — COMMENTED** [review]
It looks ok, it's just the update query it's a bit complex now to read
Also to make sure:
Are we removing the trigger to updating the score in favour of a higher updating frequency?

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Looks good, I noticed that you removed the `scheduleRecalcForVideo` method. Although it was not used right now, I think this could have been a very useful addition if used properly. We can have two different relevance score calculation intervals, a smaller interval (e.g. 10 mins) for all the scheduled videos (i.e. scheduled due to view, comments, channel weight updates, etc.), and a larger interval for table-wide recalculation.

Since the `video` table is growing rapidly I think recalculation of the relevance score for all the videos every 10 minutes, is not ideal, so the relevance score update for the whole table can be every 12/24 hours.

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Looks good. Just a couple of points

- Please include this change to avoid potential Postgres deadlocks (https://github.com/Joystream/orion/issues/268#issuecomment-1838647280)

- What do you think about this `We can have two different relevance score calculation intervals, a smaller interval (e.g. 10 mins) for all the scheduled videos (i.e. scheduled due to view, comments, channel weight updates, etc.), and a larger interval for table-wide recalculation.` from my [comment](https://github.com/Joystream/orion/pull/251#pullrequestreview-1760917652).

File: src/utils/VideoRelevanceManager.ts [hunk 2]
```diff
@@ -5,11 +5,35 @@ import { globalEm } from './globalEm'
 // constant used to parse seconds from creation
 export const NEWNESS_SECONDS_DIVIDER = 60 * 60 * 24
 
+type VideoRelevanceManagerLoops = {
+  fullUpdateLoopTime: number
+  scheduledUpdateLookTime: number
```
Comment (zeeshanakram3) [inline src/utils/VideoRelevanceManager.ts:10]: I think you meant `Loop` instead of `Look`

```suggestion
  scheduledUpdateLoopTime: number
```

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]

**Review by zeeshanakram3 — APPROVED** [review]
LGTM

**Comment by zeeshanakram3** [comment]
Couple of things that I missed in the review:

- Can you please schedule the recalc in the setChannelWeights mutation too (https://github.com/wradoslaw/orion/blob/1eba2c40642ac23a66656b6a15b2bc1735c71c09/src/server-extension/resolvers/AdminResolver/index.ts#L198)
- Now full table calculation is set at 12 hrs, so effectively after each deployment till 12 hours Gleev homepage wont have the correct videos since full recalculation has not happened yet. So maybe we should run the full recalc. at the start of the new deployment and after the state import? WDYT?

Would you please address these before the tomorrow's deployment. Thanks

**Comment by WRadoslaw** [comment]
> Couple of things that I missed in the review:
> 
> * Can you please schedule the recalc in the setChannelWeights mutation too (https://github.com/wradoslaw/orion/blob/1eba2c40642ac23a66656b6a15b2bc1735c71c09/src/server-extension/resolvers/AdminResolver/index.ts#L198)
> * Now full table calculation is set at 12 hrs, so effectively after each deployment till 12 hours Gleev homepage wont have the correct videos since full recalculation has not happened yet. So maybe we should run the full recalc. at the start of the new deployment and after the state import? WDYT?
> 
> Would you please address these before the tomorrow's deployment. Thanks

1. Ok.
2. It's already done [here](https://github.com/Joystream/orion/blob/1eba2c40642ac23a66656b6a15b2bc1735c71c09/src/processor.ts#L333), it doesn't show up, because this change is older than my PR

**Pull request merged**